### ISSUE TITLE
NativeArrays returned from ARFace's GetUndisposable throw NullReferenceException when accessing an element in the Editor

### ISSUE THREAD

**Issue by Lchikar** [Unity-Technologies/arfoundation-samples#843]
Hello @tdmowrer,

I have dealt with the exact [same issue as @asus4](https://github.com/Unity-Technologies/arfoundation-samples/issues/343#issue-514009897) lately while working on an ARFoundation simulation tool as well. I managed to solve it temporarily with what @asus4 shared in [his thread](https://github.com/asus4/ARKitStreamer/issues/2#issuecomment-552880905).

Are there any updates about this issue today ?

Environment :
Unity Editor 2019.4.24f1
ARFoundation 4.1.7
ARKit 4.1.7
ARCore 4.1.7

**Comment by tdmowrer** [comment]
A fix is coming soon. The correct fix is to add an atomic safety handle when safety checks are enabled (by default, they are enabled in the Editor, but not the Player, which is why you only see this in the Editor). ARFace's `GetUndisposable` method should look like this:

```cs
static unsafe NativeArray<T> GetUndisposable<T>(NativeArray<T> disposable) where T : struct
{
    if (!disposable.IsCreated)
        return default;

    var undisposable = NativeArrayUnsafeUtility.ConvertExistingDataToNativeArray<T>(
        disposable.GetUnsafePtr(),
        disposable.Length,
        Allocator.None);

#if ENABLE_UNITY_COLLECTIONS_CHECKS
    NativeArrayUnsafeUtility.SetAtomicSafetyHandle(
        ref undisposable,
        NativeArrayUnsafeUtility.GetAtomicSafetyHandle(disposable));
#endif

    return undisposable;
}
```

**Comment by Lchikar** [comment]
Thank you for your response, I will try that.
Do you know how long until this next fix, maybe ?

**Comment by tdmowrer** [comment]
I can't be more specific than "soon", unfortunately, but it is our intention to release it as soon as possible.

**Comment by todds-unity** [comment]
This should now be released in 4.2.0-pre.12.
